merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetProfileSegmentsAction.java

```diff
@@ -50,23 +52,37 @@ public <Dynamics> List<Pair<Duration, Dynamics>> get(
     final var simulationDuration = simulationWindow.duration();
     if (resultSet.next()) {
       var offset = parseOffset(resultSet, 1, simulationStart);
-      var dynamics = getJsonColumn(resultSet, "dynamics", dynamicsP)
-          .getSuccessOrThrow(failureReason -> new Error("Corrupt profile dynamics: " + failureReason.reason()));
+      Optional<Dynamics> dynamics;
+      var isGap = resultSet.getBoolean("is_gap");
+      if (!isGap) {
+        dynamics = Optional.of(getJsonColumn(resultSet, "dynamics", dynamicsP)
+            .getSuccessOrThrow(failureReason -> new Error("Corrupt profile dynamics: " + failureReason.reason())));
+      } else {
+        dynamics = Optional.empty();
+      }
 
       while (resultSet.next()) {
         final var nextOffset = parseOffset(resultSet, 1, simulationStart);
         final var duration = nextOffset.minus(offset);
         segments.add(Pair.of(duration, dynamics));
         offset = nextOffset;
-        dynamics = getJsonColumn(resultSet, "dynamics", dynamicsP)
-            .getSuccessOrThrow(
-                failureReason -> new Error("Corrupt profile dynamics: " + failureReason.reason()));
+
+        isGap = resultSet.getBoolean("is_gap");
+        if (!isGap) {
+          dynamics = Optional.of(getJsonColumn(resultSet, "dynamics", dynamicsP)
+              .getSuccessOrThrow(
+                  failureReason -> new Error("Corrupt profile dynamics: " + failureReason.reason())));
+        } else {
+          dynamics = Optional.empty();
+        }
       }
 
       final var duration = simulationDuration.minus(offset);
       segments.add(Pair.of(duration, dynamics));
     }
```

Comment:
Wouldn't hurt to change this `}` to:
```java
    } else {
      throw new Error("No profile segments found for `dataset_id` (%d) and `profile_id` (%d)".formatted(datasetId, profileId));
    }
```
This way a legible error would be shown in this case instead of `segments.remove()` below causing an array index out of bounds exception.

Reply:
(I'm referring to line 82 here btw)